GANDLF/entrypoints/collect_stats.py

```diff
@@ -129,7 +135,78 @@ def plot_all(df_training, df_validation, df_testing, output_plot_dir):
     return df_training, df_validation, df_testing
 
 
-if __name__ == "__main__":
+def _read_data_and_plot(
+    training_logs_path: str,
+    validation_logs_path: str,
+    testing_logs_path: Optional[str],
+    output_plot_path: str,
+    output_file: str,
+):
+    # moved out from _collect_stats for easier testing
+    # Read all the files
+    df_training = pd.read_csv(training_logs_path)
+    df_validation = pd.read_csv(validation_logs_path)
+    df_testing = pd.read_csv(testing_logs_path) if testing_logs_path else None
+
+    # Check for metrics in columns and do tight plots
+    plot_all(df_training, df_validation, df_testing, output_plot_path)
+
+    df_training["split"] = "train"
+    df_testing["split"] = "test"
+    df_validation["split"] = "validation"
+    pd.concat((df_training, df_testing, df_validation)).to_csv(output_file)
+
+
+def _collect_stats(model_dir: str, output_dir: str):
+    input_dir = os.path.normpath(model_dir)
+    output_dir = os.path.normpath(output_dir)
+    Path(output_dir).mkdir(parents=True, exist_ok=True)
+    # TODO: here is bug? output_file is never used. We don't save data.csv at all?
```

Comment:
is this TODO relevant? I see `output_file` is being read

Reply:
Agreed.